### PR TITLE
Improve Karate CI to push the generated reports on a branch

### PR DESCRIPTION
**Pull request by GabrielFleischer**

### CODE REVIEW

File: .github/workflows/karate.yml [hunk 1]
```diff
@@ -55,6 +55,42 @@ jobs:
           name: Scala Server Logs
           path: ./tests/karate/scala.log
           
+      # The next actions are pushing the new report to the defined branch
+      - name: Checkout report branch
+        if: ${{ always() }} # execute even if previous steps failed
```
Comment (jbsv) [inline .github/workflows/karate.yml:60]: I doubt this one is useful

File: .github/workflows/karate.yml [hunk 2]
```diff
@@ -55,6 +55,42 @@ jobs:
...
+          rm -rf report-repo/*
+          cp -a ./tests/karate/target/cucumber-html-reports/. ./report-repo/
+          
+      - name: Get current date
+        if: ${{ always() }} # execute even if previous steps failed
+        id: date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+        
+      - name: Commit files
+        if: ${{ always() }} # execute even if previous steps failed
```
Comment (jbsv) [inline .github/workflows/karate.yml:78]: I would remove this one

File: .github/workflows/karate.yml [hunk 3]
```diff
@@ -55,6 +55,42 @@ jobs:
...
+        working-directory: ./report-repo
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git add -A
+          git commit -m "Report from ${{ steps.date.outputs.date }}"
+          
+      - name: Push changes
+        if: ${{ always() }} # execute even if previous steps failed
```
Comment (jbsv) [inline .github/workflows/karate.yml:87]: if the previous git commands fail, this one seems useless too.

File: .github/workflows/karate.yml [hunk 4]
```diff
@@ -112,3 +148,39 @@ jobs:
         with:
           name: Go Server Logs
           path: ./tests/karate/go.log
+          
+      # The next actions are pushing the new report to the defined branch
+      - name: Checkout report branch
+        if: ${{ always() }} # execute even if previous steps failed
```
Comment (jbsv) [inline .github/workflows/karate.yml:154]: same

File: .github/workflows/karate.yml [hunk 5]
```diff
@@ -112,3 +148,39 @@ jobs:
...
+          rm -rf report-repo/*
+          cp -a ./tests/karate/target/cucumber-html-reports/. ./report-repo/
+          
+      - name: Get current date
+        if: ${{ always() }} # execute even if previous steps failed
+        id: date
+        run: echo "::set-output name=date::$(date +'%Y-%m-%d')"
+        
+      - name: Commit files
+        if: ${{ always() }} # execute even if previous steps failed
```
Comment (jbsv) [inline .github/workflows/karate.yml:172]: same

File: .github/workflows/karate.yml [hunk 6]
```diff
@@ -112,3 +148,39 @@ jobs:
...
+        working-directory: ./report-repo
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git add -A
+          git commit -m "Report from ${{ steps.date.outputs.date }}"
+          
+      - name: Push changes
+        if: ${{ always() }} # execute even if previous steps failed
```
Comment (jbsv) [inline .github/workflows/karate.yml:181]: same

**Review by jbsv — CHANGES_REQUESTED** [review]
I would remove many "always" conditions

**Review by pierluca — COMMENTED** [review]
I'm a bit worried that the `always()` might backfire if some reporting steps fail.
However, we still need the reporting to run if the tests fail.
Is there something not as strong as `always()` that would still run despite the test failures ?

**Review by jbsv — APPROVED** [review]
LGTM

**Pull request merged**